### DISCUSSION TITLE
ScriptReference/AssetModificationProcessor.OnWillCreateAsset

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#610] [Notes]
# ScriptReference/AssetModificationProcessor.OnWillCreateAsset

https://docs.unity3d.com/ScriptReference/AssetModificationProcessor.OnWillCreateAsset.html

**Comment by electronics45** [comment]
Note that this method is only called when the importing asset does not yet have a valid .meta file.  I.E.  files added via vcs (where the .meta file is included), or files copied from another project along with their .meta file will not trigger this event.

**Comment by qwe321qwe321qwe321** [comment]
In 2022.2.X, OnWillCreateAsset was called for both the assetPath and the metaFilePath, whereas in 2021, it was only called for the metaFilePath. This change led to FileNotFoundExceptions in some older scripts.
Unity Forum: [FileNotFoundException when reading asset contents in OnWillCreateAsset](https://forum.unity.com/threads/filenotfoundexception-when-reading-asset-contents-in-onwillcreateasset.1399756/)
